pyeti/api.py

```diff
@@ -231,6 +231,45 @@ def observable_bulk_add(self, observables, tags=None):
         json = {"observables": [{"tags": tags, "value": o} for o in observables]}
         return self._make_post('observable/bulk', json=json)
 
+    def launch_analytics_oneshot(self, name_of_oneshot, value, type_obversable):
+        all_analytics = self.list_analytics_oneshot()
+
+        oneshot = [ item for item in all_analytics if item['name'] == name_of_oneshot]
+
+        if oneshot:
+            if type_obversable in oneshot[0]['acts_on']:
+                print(value)
+                observable = self.observable_add(value)
+                if observable:
+                    oneshot_inst = self._make_post('analytics/oneshot/%s/run' % oneshot[0]['id'], data={'id': observable['id']})
+                    status =self.analytics_oneshot_status(oneshot_inst['_id'])
+
+                    while status['status'] =='running':
+                        status = self.analytics_oneshot_status(oneshot_inst['_id'])
+                        continue
+
+                    if status['status'] == 'finished':
+                        return status['results']
+                    else:
+                        logging.error('Error Oneshot Processing %s with %s' % (name_of_oneshot, value))
```

Comment:
Details on the error would be useful (you could add a log line with the `status` variable here.